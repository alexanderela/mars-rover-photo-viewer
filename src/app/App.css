#root {
  max-width: 1280px;
  margin: 0;
  text-align: center;
  width: 100%;
}

body {
  margin: 0;
}
