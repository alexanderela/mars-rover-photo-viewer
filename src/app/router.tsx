import {
  createBrowserRouter,
  redirect,
} from "react-router-dom";
import { EnlargedPhotoViewer } from "../features/EnlargedPhotoViewer";

export const routesConfig = [
  {
    path: "/",
    lazy: async () => {
      const { default: App } = await import("./App");

      return {
        element: (
          <App />
        ),
      }
    },
    children: [
      {
        index: true,
        loader: async () => redirect("rovers/curiosity?page=1")
      },
      {
        path: "rovers/:rover",
        lazy: async () => {
          const { PhotoViewer } = await import("../features/photos/PhotoViewer");
    
          return {
            element: (
              <PhotoViewer />
            )
          }
        },
        children: [
          {
            path: "id/:id",
            element: <EnlargedPhotoViewer />
          }
        ]
      },
    ]
  },
];

export const router = createBrowserRouter(routesConfig);