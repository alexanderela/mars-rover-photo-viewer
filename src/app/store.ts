import { configureStore } from "@reduxjs/toolkit";
import roverPhotosReducer from "../reducers/roverPhotosSlice";
import pageSliceReducer from "../reducers/pageSlice";
import selectedRoverSliceReducer from "../reducers/selectedRoverSlice";

const reducer = {
  roverPhotos: roverPhotosReducer,
  page: pageSliceReducer,
  selectedRover: selectedRoverSliceReducer,
}

export const store = configureStore({ 
  reducer,
});

export type AppStore = typeof store;
export type RootState = ReturnType<AppStore['getState']>;
export type AppDispatch = AppStore['dispatch'];