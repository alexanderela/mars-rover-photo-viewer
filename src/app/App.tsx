import Grid from "@mui/material/Grid2";
import { Header } from "../layout/Header";
import { SideNav } from "../layout/SideNav";
import { Content } from "../layout/Content";
import "./App.css";

const App = () => {
  return (
    <>
      <Grid
        container
        sx={{
          backgroundColor: (theme) => theme.palette.primary.main,
          height: "100%",
          width: "100vw",
          padding: 0,
        }}
      >
        <Header />
        <SideNav />
        <Content />
      </Grid>
    </>
  );
};

export default App;
