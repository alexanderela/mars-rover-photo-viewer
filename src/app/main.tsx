import { createRoot } from "react-dom/client";
import { Provider } from "react-redux";
import { ThemeProvider } from "@mui/material/styles";
import { theme } from "../styles/theme.ts";
import { setupStore } from "./store.ts";
import { RouterProvider } from "react-router-dom";
import { router } from "./router.tsx";
import "../index.css";
import "../reset.css";

createRoot(document.getElementById("root")!).render(
  <Provider store={setupStore()}>
    <ThemeProvider theme={theme}>
      <RouterProvider router={router} />
    </ThemeProvider>
  </Provider>,
);
