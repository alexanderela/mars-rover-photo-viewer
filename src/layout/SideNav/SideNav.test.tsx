import { describe, expect, it } from "vitest";
import { render, waitFor } from "../../tests/test-utils";
import { createMemoryRouter, RouterProvider } from "react-router-dom";
import { routesConfig } from "../../app/router";

describe("<SideNav", () => {
  it("renders SideNav component", async () => {
    const router = createMemoryRouter(routesConfig, {
      initialEntries: ["/", "/photos/rovers/curiosity?page=1"],
      initialIndex: 1
    });
    const result = render(<RouterProvider router={router} />)

    await waitFor(() => {
      const element = result.getByTestId(/SideNav/i);
      expect(element).toBeInTheDocument();
      expect(element).toMatchSnapshot();
    })
  });
});