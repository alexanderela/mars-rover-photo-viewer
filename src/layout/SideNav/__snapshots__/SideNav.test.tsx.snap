// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SideNav > renders SideNav component 1`] = `
<div
  class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-1 css-1ahra1l-MuiGrid2-root"
  data-testid="SideNav"
>
  <div
    class="MuiBox-root css-11gtdaf"
  >
    <button
      aria-label="rover photos"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-ty9iju-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-uks7c6-MuiSvgIcon-root"
        data-testid="PhotoLibraryIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 16V4c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2m-11-4 2.03 2.71L16 11l4 5H8zM2 6v14c0 1.1.9 2 2 2h14v-2H4V6z"
        />
      </svg>
    </button>
    <p
      class="MuiTypography-root MuiTypography-body2 css-1elpnu6-MuiTypography-root"
    >
      Rover Photos
    </p>
  </div>
  <div
    class="MuiBox-root css-11gtdaf"
  >
    <button
      aria-label="rover map"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-ty9iju-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
      value="map"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-uks7c6-MuiSvgIcon-root"
        data-testid="MapIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m20.5 3-.16.03L15 5.1 9 3 3.36 4.9c-.21.07-.36.25-.36.48V20.5c0 .28.22.5.5.5l.16-.03L9 18.9l6 2.1 5.64-1.9c.21-.07.36-.25.36-.48V3.5c0-.28-.22-.5-.5-.5M15 19l-6-2.11V5l6 2.11z"
        />
      </svg>
    </button>
    <p
      class="MuiTypography-root MuiTypography-body2 css-1elpnu6-MuiTypography-root"
    >
      Rover Map
    </p>
  </div>
  <div
    class="MuiBox-root css-11gtdaf"
  >
    <button
      aria-label="favorite photos"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-ty9iju-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
      value="favorites"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-uks7c6-MuiSvgIcon-root"
        data-testid="FavoriteIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54z"
        />
      </svg>
    </button>
    <p
      class="MuiTypography-root MuiTypography-body2 css-1elpnu6-MuiTypography-root"
    >
      Favorites
    </p>
  </div>
</div>
`;
